### PR TITLE
feat(api): Make gripper errors nonfatal

### PR DESCRIPTION
**Pull request by SyntaxColoring**
## Overview

Closes EXEC-714.

## Test Plan and Hands on Testing

* [x] Physically try various ways of interfering with a gripper movement, and make sure they all cause the run to enter recovery mode now.
    * [x] Missing labware at source
    * [x] Obstacle at destination causing stall/collision
    * [x] Gripper removed from jaw in transit 

## Changelog

Catch some exceptions in the gripper part of `MoveLabwareImplementation` and turn them into a new defined error, `errorType: "gripperMovement"`.

## Review requests

See comments below.

## Risk assessment

Medium.

### CODE REVIEW

File: api/src/opentrons/protocol_engine/execution/labware_movement.py [hunk 1]
```diff
@@ -177,6 +177,9 @@ async def move_labware_with_gripper(
                             labware_id
                         )
                         well_bbox = self._state_store.labware.get_well_bbox(labware_id)
+                        # todo(mm, 2024-09-26): This currently raises a lower-level 2015 FailedGripperPickupError.
+                        # Convert this to a higher-level 3001 LabwareDroppedError or 3002 LabwareNotPickedUpError,
+                        # depending on what waypoint we're at, to propagate a more specific error code to users.
```
Comment (SyntaxColoring) [inline api/src/opentrons/protocol_engine/execution/labware_movement.py:182]: Alternatively, we could consolidate 3001 and 3002.

File: api/src/opentrons/protocol_engine/commands/move_labware.py [hunk 2]
```diff
@@ -171,6 +197,18 @@ async def execute(  # noqa: C901
...
+        # pipette is no longer over any labware. This is necessary for safe path
+        # planning in case the next movement goes to the same labware (now in a new
+        # place).
+        pipette_location = self._state_view.pipettes.get_current_location()
+        if (
+            isinstance(pipette_location, CurrentWell)
+            and pipette_location.labware_id == params.labwareId
+        ):
+            state_update.clear_all_pipette_locations()
+
```
Comment (SyntaxColoring) [inline api/src/opentrons/protocol_engine/commands/move_labware.py]: Oops, this duplication looks like a merge error. Need to fix this before merging.

File: api/src/opentrons/protocol_engine/commands/move_labware.py [hunk 3]
```diff
@@ -76,27 +88,41 @@ class MoveLabwareResult(BaseModel):
...
 
 
-class MoveLabwareImplementation(
-    AbstractCommandImpl[MoveLabwareParams, SuccessData[MoveLabwareResult, None]]
-):
+class GripperMovementError(ErrorOccurrence):
+    """Returned when something physically goes wrong when the gripper moves labware.
+
+    When this error happens, the engine will leave the labware in its original place.
+    """
```
Comment (SyntaxColoring) [inline api/src/opentrons/protocol_engine/commands/move_labware.py:95]: When this error happens, *do* we want the engine to leave the labware in its original place?

We could also, for example, move it off-deck.

File: api/src/opentrons/protocol_engine/commands/aspirate.py [hunk 4]
```diff
@@ -161,7 +161,7 @@ async def execute(self, params: AspirateParams) -> _ExecuteReturn:
             )
 
 
-class Aspirate(BaseCommand[AspirateParams, AspirateResult, ErrorOccurrence]):
+class Aspirate(BaseCommand[AspirateParams, AspirateResult, OverpressureError]):
```
Comment (SyntaxColoring) [inline api/src/opentrons/protocol_engine/commands/aspirate.py:164]: Unrelated fix, just making sure `OverpressureError` shows up in these commands' OpenAPI specs.

File: api/src/opentrons/protocol_engine/commands/move_labware.py (hunk 3 above)
Comment (SyntaxColoring) [reply api/src/opentrons/protocol_engine/commands/move_labware.py:95]: @mjhuff

**Review by TamarZanzouri — APPROVED** [review]
LGTM! thank you!

**Comment by SyntaxColoring** [comment]
The failing snapshot test is an `edge` change, unrelated to this PR.

File: api/src/opentrons/protocol_engine/commands/move_labware.py (hunk 3 above)
Comment (mjhuff) [reply api/src/opentrons/protocol_engine/commands/move_labware.py:95]: @SyntaxColoring Thanks for asking. Leaving this in its original location would be significantly easier to work with from a FE perspective.
